tests/integration/components/flash-message-test.js

```diff
@@ -1,144 +1,172 @@
+    
+    await settled();
+
+    assert.dom('.alert').hasClass('active', 'it sets `active` to true after rendering');
   });
-});
 
-test('flash message is removed after timeout', function(assert) {
-  assert.expect(3);
+  test('it does not error when quickly removed from the DOM', async function(assert) {
+    this.set('flash', FlashMessage.create({ message: 'hi', sticky: true }));
+    this.set('flag', true);
 
-  this.set('flash', FlashMessage.create({
-    message: 'hi',
-    sticky: false,
-    timeout: timeoutDefault
-  }));
+    await render(hbs`
+      {{#if flag}}
+        {{#flash-message flash=flash as |component flash|}}
+          {{flash.message}}
+        {{/flash-message}}
+      {{/if}}
+    `);
 
-  this.render(hbs`
-    {{#flash-message flash=flash as |component flash|}}
-      {{flash.message}}
-    {{/flash-message}}
-  `);
+    this.set('flag', false);
+
+    return settled().then(() => {
+      assert.ok(this.get('flash').isDestroyed, 'Flash Object isDestroyed');
+    });
+  });
 
-  assert.equal(this.$().text().trim(), 'hi');
-  assert.notOk(this.get('flash').isDestroyed, 'Flash is not destroyed immediately');
+  test('flash message is removed after timeout', async function(assert) {
+    assert.expect(3);
+
+    this.set('flash', FlashMessage.create({
+      message: 'hi',
+      sticky: false,
+      timeout: timeoutDefault
+    }));
+
+    render(hbs`
+      {{#flash-message flash=flash as |component flash|}}
+        {{flash.message}}
+      {{/flash-message}}
+    `);
+
+    next(this, () => {
+      assert.dom('*').hasText('hi');
+      assert.notOk(this.get('flash').isDestroyed, 'Flash is not destroyed immediately');
+    });
+
+    await settled();
 
-  return wait().then(() => {
     assert.ok(this.get('flash').isDestroyed, 'Flash Object is destroyed');
   });
-});
 
-test('flash message is removed after timeout if mouse enters', function(assert) {
-  assert.expect(3);
+  test('flash message is removed after timeout if mouse enters', async function(assert) {
+    assert.expect(3);
 
-  let flashObject = FlashMessage.create({
-    message: 'hi',
-    sticky: false,
-    timeout: timeoutDefault
-  });
+    let flashObject = FlashMessage.create({
+      message: 'hi',
+      sticky: false,
+      timeout: timeoutDefault
+    });
+
+    this.set('flash', flashObject);
 
-  this.set('flash', flashObject);
+    render(hbs`
+      {{#flash-message elementId="testFlash" flash=flash as |component flash|}}
+        {{flash.message}}
+      {{/flash-message}}
+    `);
 
-  this.render(hbs`
-    {{#flash-message elementId="testFlash" flash=flash as |component flash|}}
-      {{flash.message}}
-    {{/flash-message}}
-  `);
+    next(this, () => {
+      assert.dom('*').hasText('hi');
+      triggerEvent('#testFlash', 'mouseenter');
 
-  assert.equal(this.$().text().trim(), 'hi');
-  this.$('#testFlash').mouseenter();
+      next(this, () => {
+        assert.notOk(flashObject.isDestroyed, 'Flash Object is not destroyed');
```

Comment:
Looks like doing this solves 3.x but not 2.x. Might have to write a workaround of using the `later` run loop. However, I'm unsure if that is a right approach.

Reply:
Hmm, yeah it's hard to tell if the appropriate `timeout` has passed here.  Are we getting a false positive?  If we don't do a `mouseenter` above, does this still pass?

Would it make sense to do `await settled()` earlier?

i.e.
`render -> next -> mouseenter -> settled() -> assert.notOk(isDestroyed) -> mouseleave -> next -> assert(isDestroyrd)`?

Reply:
> Are we getting a false positive? 
Nope, It is not a false positive.

> Would it make sense to do await settled() earlier?
- I wouldn't do so since the `await settled` is used to wait for all the animation timers to be complete if I'm not wrong. I tried the order you had mentioned for 3.4, and it fails in the `assert.notOk(flashObject.isDestroyed, 'Flash Object is not destroyed');` test case.

```
next(this, async () => {
  triggerEvent('#testFlash', 'mouseenter');
  await settled();

      assert.notOk(flashObject.isDestroyed, 'Flash Object is not destroyed');

      triggerEvent('#testFlash', 'mouseleave');

      next(this, () => {
        assert.ok(flashObject.isDestroyed, 'Flash Object is destroyed');
      });
    });
```